test/renderer/image_manager.test.cpp

```diff
@@ -108,12 +108,21 @@ TEST(ImageManager, RemoveReleasesBinPackRect) {
 
 class StubImageRequestor : public ImageRequestor {
 public:
+    StubImageRequestor(ImageManager* imageManager_ = nullptr) : imageManager(imageManager_) {}
+
     void onImagesAvailable(ImageMap icons, ImageMap patterns, std::unordered_map<std::string, uint32_t> versionMap, uint64_t imageCorrelationID_) final {
         if (imagesAvailable && imageCorrelationID == imageCorrelationID_) imagesAvailable(icons, patterns, versionMap);
     }
 
+    ~StubImageRequestor() {
+        if (imageManager) {
```

Comment:
Shouldn't this functionality be the part of the `ImageRequestor` class? Apparently, `ImageRequestor` is not quite functional without `mageManager`, so it could even accept a reference to it and do without null-checking in the destructor.